instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb

```diff
@@ -21,7 +22,7 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         compatible do
-          gem_version >= MINIMUM_VERSION
+          gem_version >= MINIMUM_VERSION && gem_version < MAXIMUM_VERSION
```

Comment:
💁🏼 In my adventures with adding the default gem compatibility, this functionality is provided to us by  [`Gem::Requirement#satisfied_by?`](https://ruby-doc.org/stdlib-3.0.3/libdoc/rubygems/rdoc/Gem/Requirement.html) and may be a reasonable alternative: 

```ruby
    irb(main):028:0> r = Gem::Requirement.new(">= 5.2.0", "< 7.0.0")
=> #<Gem::Requirement:0x00007ff68714ea30 @requirements=[[">=", #<Gem::Version "5.2.0">], ["<", #<Gem::Version "7.0.0">]]>
irb(main):029:0> r.satisfied_by?(Gem::Version.new("5.1.9"))
=> false
irb(main):030:0> r.satisfied_by?(Gem::Version.new("5.2"))
=> true
irb(main):031:0> r.satisfied_by?(Gem::Version.new("6.9"))
=> true
irb(main):032:0> r.satisfied_by?(Gem::Version.new("7"))
=> false
irb(main):033:0> r.satisfied_by?(Gem::Version.new("7.0"))
=> false
irb(main):034:0> r.satisfied_by?(Gem::Version.new("7.0.0.alpha1"))
=> true
```

Reply:
Maybe not so funky? I guess it is technically true?
```
irb(main):036:0>  Gem::Version.new("7.0.0.alpha1") < Gem::Version.new("7.0.0")
=> true
```

Reply:
oh, hm weird, good point @arielvalentin . this doesn't apply to most libs, but for this use case, maybe `::ActiveRecord::VERSION::MAJOR < 7` ([defined here](https://github.com/rails/rails/blob/8a419e8d4d7e15a3b25344d4b727e6410404f093/activerecord/lib/active_record/gem_version.rb#L10)) may work better for us here

Reply:
Good callout, this is a case that is very relevant to the value of this PR.  

We can drop the `.alpha/whaterver` from the versioning or check for its presence using release/prerelease?.  But I think I can rework this to be a bit clearer/simpler.

I'll update the PR in a moment.

Reply:
Updated.

Reply:
lgtm